### PR TITLE
Allow use of private endpoints for Vertex AI requests

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds handling for a subnetwork kwarg, allowing the use of a private Vertex AI endpoint and uncapping the maximum request size the service will accept. Also updates the docstring to reflect this. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/ml/inference/vertex_ai_inference.py [hunk 1]
```diff
@@ -83,9 +87,15 @@ def __init__(
     """
 
     self._env_vars = kwargs.get('env_vars', {})
+
     # TODO: support the full list of options for aiplatform.init()
     # See https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform#google_cloud_aiplatform_init
-    aiplatform.init(project=project, location=location, experiment=experiment)
+    subnetwork = kwargs.get('subnetwork', None)
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/ml/inference/vertex_ai_inference.py:93]: I'm a little torn on accepting this as a kwarg or making it an optional named param to the init function

Comment (jrmccluskey) [reply sdks/python/apache_beam/ml/inference/vertex_ai_inference.py]: Went ahead and made it an optional positional arg instead while making other fixes as well

File: sdks/python/apache_beam/ml/inference/vertex_ai_inference.py [hunk 2]
```diff
@@ -66,26 +66,35 @@ def __init__(
...
+    IO. Public Vertex AI endpoints have a maximum request size of 1.5 MB.
+    If you wish to make larger requests and use a private endpoint, provide
+    the Compute Engine network you wish to use.
+
     Args:
       endpoint_id: the numerical ID of the Vertex AI endpoint to query
       project: the GCP project name where the endpoint is deployed
       location: the GCP location where the endpoint is deployed
       experiment (Optional): experiment label to apply to the queries
+      network (Optional): the full name of the Compute Engine network
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/inference/vertex_ai_inference.py:86]: just to make sure, this is network and not subnetwork? https://cloud.google.com/vpc/docs/vpc#vpc_networks_and_subnets

Would it make sense to add an example akin to: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/options/pipeline_options.py#L1045-L1046

**Comment by tvalentyn** [comment]
Not sure why PyDocs complaints, seems strange as similar pattern is present. I'd try to run locally to undestand what exactly triggers it..

File: sdks/python/apache_beam/ml/inference/vertex_ai_inference.py (hunk 2 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/ml/inference/vertex_ai_inference.py:86]: This is network and not subnetwork, following the aiplatform documentation (https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform#google_cloud_aiplatform_init). 

Yeah the docstring getting an example makes sense, let me do that

Comment (jrmccluskey) [reply sdks/python/apache_beam/ml/inference/vertex_ai_inference.py]: Done

**Comment by jrmccluskey** [comment]
@tvalentyn this should be ready now

**Comment by tvalentyn** [comment]
LGTM

**Pull request merged**